### PR TITLE
Empty pallet location

### PR DESCRIPTION
**Pull request by lmatoti**
Summary of this change:

Empty pallet location

An RMD app that allows the user to move all pallets in a location to 'PENDING_LOCATION'

CHANGELOG entry:

**Added**

- New 'Empty Pallet Location' application

How to test this code via the UI:
RMD | Finished Goods | Empty Pallet Location

1. Create 'PENDING_LOCATION' via the seeds
2. In the 'Empty Pallet Location' app, scan a location app will display the number of pallets in location and ask you to confirm if you want this location to be emptied.
3. Click 'yes' and all pallets will be moved out of this location

### CODE REVIEW

File: lib/masterfiles/repositories/location_repo.rb [hunk 1]
```diff
@@ -414,6 +414,13 @@ def for_select_location_for_assignment(assignment_code)
         .map(%i[location_short_code id])
     end
 
+    def location_pallets_count(location_id)
+      DB[:locations]
+        .join(:pallets, location_id: :id)
+        .where(location_id: location_id)
+        .count
+    end
```
Comment (jcsjcs) [inline lib/masterfiles/repositories/location_repo.rb:422]: Couldn't this just be
```ruby
DB[:pallets].where(location_id: location_id).count
```

File: routes/rmd/finished_goods.rb [hunk 2]
```diff
@@ -876,6 +876,68 @@ class Nspack < Roda
...
+                                         scan_with_camera: @rmd_scan_with_camera,
+                                         caption: 'Location Emptied',
+                                         reset_button: false,
+                                         no_submit: true,
+                                         action: '',
+                                         button_caption: '')
+          form.add_section_header(res.message)
+          form.add_csrf_tag csrf_tag
+          view(inline: form.render, layout: :layout_rmd)
+        else
```
Comment (jcsjcs) [inline routes/rmd/finished_goods.rb:936]: You're just showing a message to the user, couldn't you rather do something like:
```ruby
view(inline: rmd_success_message(res.message), layout: :layout_rmd)
```

File: lib/finished_goods/interactors/pallet_movements_interactor.rb [hunk 3]
```diff
@@ -89,6 +89,33 @@ def set_pallets_target_customer(target_customer_id, pallet_ids)
       success_response("Selected pallets have been successfully allocated to target customer #{get_target_customer_name(target_customer_id)}")
     end
 
+    def empty_pallet_location(params)
```
Comment (jcsjcs) [inline lib/finished_goods/interactors/pallet_movements_interactor.rb:92]: Perhaps rename this method to `pallets_in_location_to_be_cleared`.

**Review by jcsjcs — CHANGES_REQUESTED** [review]

**Comment by lmatoti** [comment]
Changes implemented as per code review.

**Pull request merged**